public/components/notebooks/components/paragraph_components/paragraphs.tsx

```diff
@@ -160,45 +166,78 @@ export const Paragraphs = forwardRef((props: ParagraphProps, ref) => {
   }));
 
   const fetchVisualizations = async () => {
-    let opt1: EuiComboBoxOptionOption[] = [];
-    let opt2: EuiComboBoxOptionOption[] = [];
-    await http
-      .get(`${NOTEBOOKS_API_PREFIX}/visualizations/${dataSourceMDSId ?? ''}`)
-      .then((res) => {
-        opt1 = res.savedVisualizations.map((vizObject) => ({
-          label: vizObject.label,
-          key: vizObject.key,
-          className: 'VISUALIZATION',
-        }));
-      })
-      .catch((err) => console.error('Fetching dashboard visualization issue', err.body.message));
+    if (dataSourceEnabled) {
+      let opts: EuiComboBoxOptionOption[] = [];
+      const vizOptions: SavedObjectsFindOptions = {
+        type: 'visualization',
+        ...(coreRefs.application?.capabilities?.workspaces?.enabled
+          ? { workspaces: coreRefs.workspaces?.currentWorkspace$.getValue()?.name }
+          : {}),
+      };
+      await coreRefs.savedObjectsClient
+        ?.find(vizOptions)
+        .then((res) => {
+          opts = res.savedObjects.map((vizObject) => ({
+            label: vizObject.attributes.title,
+            key: vizObject.id,
+            className: 'VISUALIZATION',
+          }));
+        })
+        .catch((error) => {
+          console.error('Issue is fetching visualizations', error);
+        });
+
+      const allVisualizations = [{ label: 'Dashboards Visualizations', options: opts }];
+      setVisOptions(allVisualizations);
 
-    await SavedObjectsActions.getBulk<ObservabilitySavedVisualization>({
-      objectType: ['savedVisualization'],
-    })
-      .then((res) => {
-        opt2 = res.observabilityObjectList
-          .filter((visualization) => !visualization.savedVisualization.application_id)
-          .map((visualization) => ({
-            label: visualization.savedVisualization.name,
-            key: visualization.objectId,
-            className: 'OBSERVABILITY_VISUALIZATION',
+      const selectedObject = filter([...opts], {
+        key: para.visSavedObjId,
+      });
+      if (selectedObject.length > 0) {
+        setVisType(selectedObject.className);
```

Comment:
selectedObject[0].className?

what about other elements, do they have the same className?

Reply:
yes it should be with array index, thanks updated!